### DISCUSSION TITLE
FEniCS installation

### DISCUSSION THREAD

**Discussion by claracoletti** [msolides-2023/MU5MES01-2023#5] [Q&A]
Bonjour,
Après de nombreuses tentatives pour l'installation de FEniCS via Ubuntu, je bloque toujours à l'étape 5 qui charge indéfiniment dans la partie "Solving environment". J'ai également essayé la partie Troubleshooting à la fin du document d'installation mais l'étape 2 a aussi conduit à une erreur. Pourriez-vous me rappeler quelles étapes je peux faire afin de résoudre ce problème ?

**Comment by cmaurini** [comment]
Merci @claracoletti pour le message collectif.

essayons par étape et regardons ou ca va bloquer. 

1. Créons un nouveau environment, activer et lui dire d'utiliser conda-forge comme source des paquets
```
mamba create --name fenicsx-test
mamba activate fenicsx-test
conda config --add channels conda-forge
conda config --set channel_priority strict
```

2. Installons les paquets de base. 
```
mamba install fenics-dolfinx matplotlib sympy scipy pip 
```

Pourriez-vous confirmer si "jusqu'ici tout va bien"?

**Reply by claracoletti** [reply]
Dès la première commande de l'étape 1, j'ai l'erreur suivante qui s'affiche:
`mamba: error: unrecognized arguments: --name fenicsx-test`

**Reply by cmaurini** [reply]
avez vous fait `mamba create --name fenicsx-test`?

**Reply by claracoletti** [reply]
Oui, c'est justement cette commande qui entraîne l'erreur précédente.

**Reply by cmaurini** [reply]
essayez `mamba create -n fenicsx-test` et si ca marche pas `conda create --name fenicsx-test`?

**Comment by cmaurini** [comment]
Autrement, je viens d'eliminer pyvista qui est probabalment la source des problèmes d'installation. Vous pouvez aussi réessayer avec les instructions sur le `Install.md`. Si déjà cela marche on installe pyvista après

**Reply by claracoletti** [reply]
Avec le nouveau fichier fenicsx-0.6.0.yml et les commandes mamba, l'installation a fonctionné sur mon ordinateur. Merci beaucoup.

**Reply by cmaurini** [reply]
et vous pouvez faire tourner l'example? Car normalement vous n'avez pas `pyvista` encore.

Sur ubuntu, vous pouvez essayer d'installer pyvista avec:

```
apt-get install python3-numpy python3-imageio python3-pil python3-appdirs python3-vtk9 -yq
pip install --no-deps scooby pyvista
```

Ajoutez `sudo ` avant s'il faut

**Reply by claracoletti** [reply]
![Resultat-code-poisson](https://github.com/msolides-2023/MU5MES01-2023/assets/145424196/469c5fb4-8d72-456f-8abf-94e35861c28d)
Pour l'instant, sans pyvista, j'obtiens ce message mais le code fonctionne.

**Reply by claracoletti** [reply]
L'installation de pyvista a bien fonctionné avec les instructions précédentes. Toutefois, j'ai ce message qui apparaît quand je lance demo_poisson.py

![avec-pyvista](https://github.com/msolides-2023/MU5MES01-2023/assets/145424196/9878a087-07df-4948-92f2-e295fa46b5b6)
